pkg/controller/controller_utils.go

```diff
@@ -95,7 +95,9 @@ var UpdateLabelBackoff = wait.Backoff{
 }
 
 var (
-	KeyFunc = cache.DeletionHandlingMetaNamespaceKeyFunc
+	KeyFunc         = cache.DeletionHandlingMetaNamespaceKeyFunc
+	podPendingFirst = map[v1.PodPhase]int{v1.PodPending: 0, v1.PodUnknown: 1, v1.PodRunning: 2}
+	podRunningFirst = map[v1.PodPhase]int{v1.PodRunning: 0, v1.PodUnknown: 1, v1.PodPending: 2}
```

Comment:
You could use the same map, right?  You could delete `podRunningFirst` and use `podPendingFirst` with `>` instead of `<` in `(s ByLogging) Less`.  But maybe using separate maps is more readable—I'm not always the best judge of readability.

Reply:
Reflecting the order with different maps allows the reader to know clearly the intention.
I think it is more readable than switch the direction of comparison.